src/ert/scheduler/lsf_driver.py

```diff
@@ -283,5 +331,56 @@ async def _process_job_update(self, job_id: str, new_state: AnyJob) -> None:
                 del self._iens2jobid[iens]
             await self.event_queue.put(event)
 
+    async def _poll_once_by_bhist(self, missing_job_ids: Iterable[str]) -> _Stat:
+        if not missing_job_ids:
+            return _Stat(**{"jobs": {}})
+
+        if time.time() - self._bhist_cache_timestamp < self._bhist_required_cache_age:
+            return _Stat(**{"jobs": {}})
+
+        process = await asyncio.create_subprocess_exec(
+            self._bhist_cmd,
+            *[str(job_id) for job_id in missing_job_ids],
+            stdout=asyncio.subprocess.PIPE,
+            stderr=asyncio.subprocess.PIPE,
+        )
+        stdout, stderr = await process.communicate()
+        if process.returncode:
```

Comment:
Is this correct?
`process.returncode` is None only when the process has not terminated yet. 
The code in this block will be executed also on success, not only on failure.

Reply:
Since we await for `process.communicate()` we will not get here until it is finished, and then we are testing for non-zero-ness for the `int`.